src/sql/azdata.d.ts

```diff
@@ -2406,6 +2406,12 @@ declare module 'azdata' {
 		 */
 		export function getAccountSecurityToken(account: Account, tenantId: string, resource: AzureResource): Thenable<AccountSecurityToken | undefined>;
 
+		/**
+		 * Encrypts access token retrieved for MSAL library.
+		 * @param token Access token to be encrypted.
+		 */
+		export function encryptAccountSecurityToken(token: AccountSecurityToken): Promise<string>;
```

Comment:
These should go in azdata.proposed.d.ts. 

I'm curious about the design though, why does the provider itself not handle the encryption/decryption and storage of the tokens? The core shouldn't need to know anything about that at all - it just gets the token from the provider which can then do the check itself to see if it needs to request a new one or can just return the cached one.

Reply:
Answered later, but if the token doesn't arrive in 'account' when `getAccountSecurityToken` is called, cache will not work. And because account reference is not shared between azure core extension and ADS core, the caller (ADS core) needs to keep their copy of 'account' up to date with encrypted token in token map.

Reply:
Internally would mean in-memory and azurecore local, so there are disadvantages:
- Tokens won't persist beyond active session, and restarting ADS would mean losing tokens immediately - saving to momento saves them globally across tool versions, so does reduce AAD calls there itself.
- Local cache copy would have to be maintained everytime a new account is added/removed or cache is cleaned, so adds maintenance overhead.